code/game/turfs/closed/wall/window_frames.dm

```diff
@@ -54,6 +59,103 @@
 
 	return FALSE
 
+///Called by the climbable element if you try climb up. Better hope you're well protected against shocks! XD
+/obj/structure/window_frame/proc/on_try_climb(mob/climber)
+	shock(climber, 100)
+
+///Gives the user a shock if they get unlucky (Based on shock chance)
+/obj/structure/window_frame/proc/shock(mob/user, shock_chance)
+	if(!has_grille) // no grille? dont shock.
+		return FALSE
+	if(!prob(shock_chance))
+		return FALSE
+	if(!in_range(src, user))//To prevent TK and mech users from getting shocked
+		return FALSE
+	var/turf/my_turf = get_turf(src)
+	var/obj/structure/cable/underlaying_cable = my_turf.get_cable_node()
+	if(underlaying_cable)
```

Comment:
Check `!underlaying_cable` at the top of the proc for an early return, saves some time